src/user-management/user-management.ts

```diff
@@ -294,6 +314,158 @@ export class UserManagement {
     return deserializeAuthenticationResponse(data);
   }
 
+  async authenticateWithSessionCookie({
+    sessionCookie,
+    cookiePassword = process.env.WORKOS_COOKIE_PASSWORD,
+  }: SessionCookieOptions): Promise<
+    | AuthenticateWithSessionCookieSuccessResponse
+    | AuthenticateWithSessionCookieFailedResponse
+  > {
+    if (!cookiePassword) {
+      throw new Error('Cookie password is required');
+    }
+
+    if (!this.jwks) {
+      throw new Error('Must provide clientId to initialize JWKS');
+    }
+
+    if (!sessionCookie) {
+      return { authenticated: false, reason: 'no_session_cookie_provided' };
+    }
+
+    const session = await unsealData<SessionCookieData>(sessionCookie, {
+      password: cookiePassword,
+    });
+
+    if (!session.accessToken) {
+      return { authenticated: false, reason: 'invalid_session_cookie' };
+    }
+
+    if (!(await this.isValidJwt(session.accessToken))) {
+      return { authenticated: false, reason: 'invalid_jwt' };
+    }
+
+    const {
+      sid: sessionId,
+      org_id: organizationId,
+      role,
+      permissions,
+    } = decodeJwt<AccessToken>(session.accessToken);
+
+    return {
+      authenticated: true,
+      sessionId,
+      organizationId,
+      role,
+      permissions,
+    };
+  }
+
+  private async isValidJwt(accessToken: string): Promise<boolean> {
+    if (!this.jwks) {
+      throw new Error('Must provide clientId to initialize JWKS');
+    }
+
+    try {
+      await jwtVerify(accessToken, this.jwks);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async refreshAndSealSessionData({
+    sessionCookie,
+    cookiePassword = process.env.WORKOS_COOKIE_PASSWORD,
+  }: SessionCookieOptions): Promise<RefreshAndSealSessionDataResponse> {
```

Comment:
New method here: this accepts an existing session cookie, refreshes the session with the refresh token, and returns a new sealed cookie. How do we feel about the overall interface here? Do we feel it provides a good level of abstraction?